lib/routes/developers/activate.js

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const auth = require('../../auth');
+const db = require('../../db');
+const hex = require('buf').to.hex;
+
+/*jshint camelcase: false*/
+module.exports = {
+  auth: {
+    strategy: auth.AUTH_STRATEGY,
+    scope: [auth.SCOPE_CLIENT_MANAGEMENT]
+  },
+  handler: function activateRegistration(req, reply) {
+    var email = req.auth.credentials.email;
+
+    return db.getDeveloper(email)
+      .then(
+        function(developer) {
+          if (developer) {
+            developer.developerId = hex(developer.developerId);
+            return reply(developer).code(200);
+          }
+        },
+        reply
+      ).then(function() {
+        return db.activateDeveloper(email);
+      }).then(function() {
+        return db.getDeveloper(email);
+      }).done(function(developer) {
+        developer.developerId = hex(developer.developerId);
```

Comment:
I _think_ this can poison the JSON DB. It returns the actual object, and modifying a property here will change it in the DB also.